### PR TITLE
Safari: Fix how ATB and version are passed to extension pages

### PR DESCRIPTION
**Pull request by andrey-p**
**Reviewer:** @jdorweiler 

**CC:** @MariagraziaAlastra 
**Depends on:** Will cause a conflict with #227, I'll resolve when either gets merged

## Description:

This PR does two things:

1. Fixes settings (specifically ATB params) not being retrieved correctly from extension pages on Safari
2. Starts retrieving extension version via `fetch` and the background process, since extension pages don't have access to the version in Safari

## Steps to test this PR:

1. Add e.g. `return console.log(this)` where the feedback form is being submitted: https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/d0046ef7723d2a4974d890155ecdcd096706fd5e/shared/js/ui/models/feedback-form.es6.js#L35
2. Go to the feedback page in both Safari and Chrome with console log open
3. Press "submit" and verify that both ATB and extension version are set correctly
4. For good measure, verify that the options page works correctly as well

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/background/chrome-events.es6.js [hunk 1]
```diff
@@ -130,6 +131,11 @@ chrome.runtime.onMessage.addListener((req, sender, res) => {
         return true
     }
 
+    if (req.getExtensionVersion) {
+        res(browserWrapper.getExtensionVersion())
+        return true
+    }
```
Comment (andrey-p) [inline shared/js/background/chrome-events.es6.js]: It's not strictly necessary to send this via the background in Chrome/FF, but I want to keep a consistent-ish API.

File: shared/js/background/safari-events.es6.js [hunk 2]
```diff
@@ -143,13 +145,25 @@ let getSetting = (e) => {
 
     let setting = JSON.parse(JSON.stringify(settings.getSetting(name)))
 
-    // Safari optons page has to send a message to the background
-    // and includes an id to help identify the correct response
-    setting.id = e.message.id
```
Comment (andrey-p) [inline shared/js/background/safari-events.es6.js]: Turns out this was erroring out in Safari. `setting` is a string here, so setting properties on it breaks.

File: shared/js/ui/base/safari-ui-wrapper.es6.js [hunk 3]
```diff
@@ -9,47 +9,72 @@ if (safari &&
...
 
 let fetch = (message) => {
     return new Promise((resolve, reject) => {
         console.log(`Safari Fetch: ${JSON.stringify(message)}`)
         if (context === 'popup') {
             safari.extension.globalPage.contentWindow.message(message, resolve)
-        } else if (context === 'options') {
-            sendOptionsMessage(message, resolve, reject)
+        } else if (context === 'extensionPage') {
+            sendExtensionPageMessage(message, resolve, reject)
```
Comment (andrey-p) [inline shared/js/ui/base/safari-ui-wrapper.es6.js]: Quick renaming pass so the names still apply.

File: shared/js/background/chrome-events.es6.js (hunk 1 above)
Comment (jdorweiler) [reply shared/js/background/chrome-events.es6.js]: 👍

File: shared/js/ui/base/safari-ui-wrapper.es6.js [hunk 4]
```diff
@@ -9,47 +9,73 @@ if (safari &&
...
 
-let closePopup = () => {
-    window.safari.self.hide()
+        let pendingResolve = pendingMessages[e.message.id]
+
+        if (!pendingResolve) { return }
+
+        delete pendingMessages[e.message.id]
+        pendingResolve(e.message.data)
+    }, true)
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply shared/js/ui/base/safari-ui-wrapper.es6.js]: Nice, I like this.

File: unit-test/ui/models/feedback-form.es6.js [hunk 5]
```diff
@@ -75,10 +79,13 @@ describe('submit', () => {
...
             browser: 'Chrome',
-            browserVersion: '60.5',
-            extensionVersion: '2018.5.1'
+            browserVersion: '60.5'
         })
         feedbackForm.updateCanSubmit()
+
+        // even though we've got spies for the properties we get via fetch,
+        // they are retrieved asynchronously - give them some time to resolve
+        setTimeout(done, 50)
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply unit-test/ui/models/feedback-form.es6.js]: This seems okay to me.

**Comment by jdorweiler** [comment]
👍

**Pull request merged**